install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -14,6 +14,7 @@
 {{- $fragmentTracking := "true" -}}
```

Comment:
I've done some local testing and I think the patch should be the following:

```patch
@@ -14,6 +14,7 @@
 {{- $fragmentTracking := "true" -}}
 {{- $crdWaitTimeout := "5m" -}}
 {{- $defaultKubeProxyReplacement := "probe" -}}
+{{- $defaultEnableCiliumEndpointSlice := "false" -}}
 
 {{- /* Default values when 1.8 was initially deployed */ -}}
 {{- if semverCompare ">=1.8" (default "1.8" .Values.upgradeCompatibility) -}}
@@ -37,10 +38,16 @@
 {{- $defaultKubeProxyReplacement = "disabled" -}}
 {{- end -}}
 
+{{- /* Default values when 1.11 was initially deployed */ -}}
+{{- if semverCompare ">=1.11" (default "1.11" .Values.upgradeCompatibility) -}}
+{{- $defaultEnableCiliumEndpointSlice = "true" -}}
+{{- end -}}
+
 {{- $ipam := (coalesce .Values.ipam.mode $defaultIPAM) -}}
 {{- $bpfCtTcpMax := (coalesce .Values.bpf.ctTcpMax $defaultBpfCtTcpMax) -}}
 {{- $bpfCtAnyMax := (coalesce .Values.bpf.ctAnyMax $defaultBpfCtAnyMax) -}}
 {{- $kubeProxyReplacement := (coalesce .Values.kubeProxyReplacement $defaultKubeProxyReplacement) -}}
+{{- $enableCiliumEndpointSlice := (coalesce .Values.enableCiliumEndpointSlice $defaultEnableCiliumEndpointSlice) -}}
 ---
 apiVersion: v1
 kind: ConfigMap
@@ -771,7 +778,7 @@ data:
   vlan-bpf-bypass: {{ .Values.bpf.vlanBypass | join " " | quote }}
 {{- end }}
 
-{{- if .Values.enableCiliumEndpointSlice }}
+{{- if eq $enableCiliumEndpointSlice "true" }}
   enable-cilium-endpoint-slice: "true"
 {{- end }}
```

Reply:
Thanks @aanm ! Sorry I hadn't spent much time getting this correct, will just shamelessly steal your solution above.